src/builder/edit_current_application_info.rs

```diff
@@ -0,0 +1,182 @@
+    #[serde(skip_serializing_if = "Option::is_none")]
+    interactions_endpoint_url: Option<Cow<'a, str>>,
+
+    /// List of tags describing the content and functionality of the app (max of 20 characters per
+    /// tag). Max of 5 tags.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    tags: Option<Vec<String>>,
+
+    /// Event webhooks URL for the app to receive webhook events.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhooks_url: Option<Cow<'a, str>>,
+
+    /// If webhook events are enabled for the app. 1 to disable, and 2 to enable.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhook_status: Option<EventWebhookStatus>,
+
+    /// List of Webhook event types to subscribe to.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhook_types: Option<Vec<EventWebhookType>>,
+}
+
+impl<'a> EditCurrentApplicationInfo<'a> {
+    /// Creates a new builder instance with all values set to None.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the custom installation URL.
+    pub fn install_url(mut self, url: impl Into<Cow<'a, str>>) -> Self {
+        self.custom_install_url = Some(url.into());
+        self
+    }
+
+    /// Sets the description of the application.
+    pub fn description(mut self, description: impl Into<Cow<'a, str>>) -> Self {
+        self.description = Some(description.into());
+        self
+    }
+
+    /// Sets the role connections verification URL.
+    pub fn verification_url(mut self, url: impl Into<Cow<'a, str>>) -> Self {
+        self.role_connections_verification_url = Some(url.into());
+        self
+    }
+
+    /// Sets the installation parameters for the application.
+    pub fn install_params(mut self, params: InstallParams) -> Self {
+        self.install_params = Some(params);
+        self
+    }
+
+    /// Sets the integration types configuration.
+    pub fn integration_types_config(
+        mut self,
+        config: HashMap<InstallationContext, InstallationContextConfig>,
+    ) -> Self {
+        self.integration_types_config = Some(config);
+        self
+    }
+
+    /// Sets the application flags.
+    pub fn flags(mut self, flags: ApplicationFlags) -> Self {
+        self.flags = Some(flags);
+        self
+    }
+
+    /// Sets the application icon hash.
+    pub fn icon(mut self, hash: &str) -> Self {
+        if let Ok(image_hash) = hash.parse() {
+            self.icon = Some(image_hash);
+        } else {
+            println!("Error parsing image hash while constructing EditCurrentApplicationInfo");
+        }
+        self
+    }
+
+    /// Sets the cover image hash.
+    pub fn cover_image(mut self, hash: &str) -> Self {
+        if let Ok(image_hash) = hash.parse() {
+            self.cover_image = Some(image_hash);
+        } else {
+            println!("Error parsing image hash while constructing EditCurrentApplicationInfo");
+        }
+        self
+    }
```

Comment:
1. This shouldn't be taking one type, parsing it into another, then printlning on the user's behalf.
2. This shouldn't be taking ImageHash, as discord documents it as "image data", therefore it should be base64 string input.